src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,96 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [titleValue, setTitleValue] = useState('');
+  const [descriptionValue, setDescriptionValue] = useState('');
+  const [imgUrlValue, setImgUrlValue] = useState('');
+  const [imdbUrlValue, setImdbUrlValue] = useState('');
+  const [imdbIdValue, setImdbIdValue] = useState('');
+
+  const isValidFields =
+    titleValue && imgUrlValue && imdbUrlValue && imdbIdValue;
+
+  const resetForm = () => {
+    setTitleValue('');
+    setDescriptionValue('');
+    setImgUrlValue('');
+    setImdbUrlValue('');
+    setImdbIdValue('');
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (isValidFields) {
+      onAdd({
+        title: titleValue,
+        description: descriptionValue || '',
+        imgUrl: imgUrlValue,
+        imdbUrl: imdbUrlValue,
+        imdbId: imdbIdValue,
+      });
+      resetForm();
+      setCount(current => current + 1);
+    }
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form className="NewMovie" key={count} onSubmit={handleSubmit}>
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={titleValue}
+        onChange={value => setTitleValue(value)}
         required
       />
 
-      <TextField name="description" label="Description" value="" />
+      <TextField
+        name="description"
+        label="Description"
+        value={descriptionValue}
+        onChange={value => setDescriptionValue(value)}
```

Comment:
it will be the same. For all cases
```suggestion
        onChange={setDescriptionValue}
```